### DISCUSSION TITLE
API to push packages to the top of the analysis and ingestion queues

### DISCUSSION THREAD

**Discussion by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#3165] [Ideas]
We could have an API that when pinged, bumps a package to the top of the analysis/ingestion queue to let package authors let us know about new releases rather than us finding them purely by polling. Package authors could automate a ping as part of a release script.

This wasn’t directly requested [here](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/issues/3091#issuecomment-2187599557) but I’ve had this in mind for a while and it could also be used to solve that problem.

We would have to implement basic rate limiting on this per package to prevent abuse but we could do something that would let authors ping as part of a release process to get the index updated more quickly.

**Comment by mesqueeb** [comment]
@daveverwer thank you for making this discussion!
